From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Wed, 1 Feb 2017 23:48:03 +0100
Subject: [PATCH] {PRINTF} kernel/trace/trace_output.c: drop useless arg
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

gcc with -Wformat-security complains:

    kernel/trace/trace_output.c: In function ‘seq_print_sym_offset’:
    kernel/trace/trace_output.c:315:2: error: Call with a non-constant
    string [-Werror=format-security]
      trace_seq_printf(s, fmt, name);
      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    kernel/trace/trace_output.c: In function ‘seq_print_sym_short’:
    kernel/trace/trace_output.c:300:2: error: Call with a non-constant
    string [-Werror=format-security]
      trace_seq_printf(s, fmt, name);
      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
---
 kernel/trace/trace_output.c | 19 ++++++++-----------
 1 file changed, 8 insertions(+), 11 deletions(-)

diff --git a/kernel/trace/trace_output.c b/kernel/trace/trace_output.c
index 1a2b7965facb..4a221487c816 100644
--- a/kernel/trace/trace_output.c
+++ b/kernel/trace/trace_output.c
@@ -339,7 +339,7 @@ static inline const char *kretprobed(const char *name)
 #endif /* CONFIG_KRETPROBES */
 
 static void
-seq_print_sym_short(struct trace_seq *s, const char *fmt, unsigned long address)
+seq_print_sym_short(struct trace_seq *s, unsigned long address)
 {
 	char str[KSYM_SYMBOL_LEN];
 #ifdef CONFIG_KALLSYMS
@@ -350,17 +350,15 @@ seq_print_sym_short(struct trace_seq *s, const char *fmt, unsigned long address)
 	name = kretprobed(str);
 
 	if (name && strlen(name)) {
-		trace_seq_printf(s, fmt, name);
+		trace_seq_puts(s, name);
 		return;
 	}
 #endif
-	snprintf(str, KSYM_SYMBOL_LEN, "0x%08lx", address);
-	trace_seq_printf(s, fmt, str);
+	trace_seq_printf(s, "0x%08lx", address);
 }
 
 static void
-seq_print_sym_offset(struct trace_seq *s, const char *fmt,
-		     unsigned long address)
+seq_print_sym_offset(struct trace_seq *s, unsigned long address)
 {
 	char str[KSYM_SYMBOL_LEN];
 #ifdef CONFIG_KALLSYMS
@@ -370,12 +368,11 @@ seq_print_sym_offset(struct trace_seq *s, const char *fmt,
 	name = kretprobed(str);
 
 	if (name && strlen(name)) {
-		trace_seq_printf(s, fmt, name);
+		trace_seq_puts(s, name);
 		return;
 	}
 #endif
-	snprintf(str, KSYM_SYMBOL_LEN, "0x%08lx", address);
-	trace_seq_printf(s, fmt, str);
+	trace_seq_printf(s, "0x%08lx", address);
 }
 
 #ifndef CONFIG_64BIT
@@ -425,9 +422,9 @@ seq_print_ip_sym(struct trace_seq *s, unsigned long ip, unsigned long sym_flags)
 	}
 
 	if (sym_flags & TRACE_ITER_SYM_OFFSET)
-		seq_print_sym_offset(s, "%s", ip);
+		seq_print_sym_offset(s, ip);
 	else
-		seq_print_sym_short(s, "%s", ip);
+		seq_print_sym_short(s, ip);
 
 	if (sym_flags & TRACE_ITER_SYM_ADDR)
 		trace_seq_printf(s, " <" IP_FMT ">", ip);
-- 
